src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1658,6 +1658,48 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             LowerHWIntrinsicCC(node, NI_AVX_PTEST, GenCondition::UGT);
             break;
 
+        // This branch will take care of the embedded rounding compatible binary nodes.
+        case NI_AVX512F_Add:
```

Comment:
I think we may want to make this table driven via a new flag.

The rules are fairly static (works for scalars and for V512) and we have a large number of instructions it will end up applying towards.